cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/ProteusClientCoreCryptoImpl.kt

```diff
@@ -49,28 +52,34 @@ class ProteusClientCoreCryptoImpl internal constructor(
     override suspend fun openOrCreate() {
         wrapException {
             File(rootDir).mkdirs()
-            coreCrypto = CoreCrypto.deferredInit(
-                path,
-                databaseKey.value,
-                null
-            )
-            migrateFromCryptoBoxIfNecessary(coreCrypto)
-            coreCrypto.proteusInit()
-            coreCrypto
+            runBlocking {
```

Comment:
suggestion: We could just make it so `wrapException` is `inline`, instead of adding `runBlocking` blocks everywhere.

This is:

```diff
    @Suppress("TooGenericExceptionCaught")
-    private fun <T> wrapException(b: () -> T): T {
+    private inline fun <T> wrapException(b: () -> T): T {
        try {
            return b()
        } catch (e: CryptoException) {
            // TODO underlying proteus error is not exposed atm
            throw ProteusException(e.message, ProteusException.Code.UNKNOWN_ERROR)
        } catch (e: Exception) {
            throw ProteusException(e.message, ProteusException.Code.UNKNOWN_ERROR)
        }
    }
```

This way, the lambda is also inlined and can be suspendable if the caller is also suspendable.

Reply:
I did that first but `wrapException` relies on [::isInitialised](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/is-initialized.html) which can't inlined.

But also I added `runBlocking` because `coreCryptoDeferredInit` suspends,  not because `wrapException` suspends.

FYI: on develop `wrapException` is inline because it doesn't need `::isInitialised`